lib/phoenix_swagger/schema_definition.ex

```diff
@@ -0,0 +1,85 @@
+defmodule PhoenixSwagger.SchemaDefinition do
+  @moduledoc """
+  Provides a DSL to define Swagger Schemas outside of `swagger_definitions/1`. This separates the definition of a schema
+  from the placement inside of swagger.
+
+  # Example:
+
+  defmodule Example.Schemas.Book do
+    use PhoenixSwagger.SchemaDefinition
+
+    swagger_schema_definition do
+      title :string, "Title", required: true
+      author :string, "Author", required: true
+    end
+  end
+  """
+  alias PhoenixSwagger.Schema
+
+  defmacro __using__(_opts) do
+    quote do
+      import PhoenixSwagger.SchemaDefinition
+    end
+  end
+
+  @doc """
+  This macro accepts a block where the schema is defined by listing the properties using a DSL.
+  Defining a property for the schema takes 2 required parameters and one optional parameter.
+
+  The first parameter is the name of the property.
+  The second parameter is the property type.
+  The third optional parameter is an optional list of parameters accepting values for `required`, `enum`, and `format`.
+
+  These values are taken and placed into a PhoenixSwagger.Schema struct.
+
+  The PhoenixSwagger.Schema struct is returned from a `schema/0` function that is provided by this macro.
+
+  Example:
+    swagger_schema_definition do
+      full_name, :string, "Full name"
+      title, :string, "Title", required: true
+      genre, :string, "Genre", enum: [:scifi, :horror, :drama, :comedy]
+      birthday, :datetime, "Birthday", format: :datetime
+    end
+  """
+  defmacro swagger_schema_definition(properties_block) do
```

Comment:
I think we can get away with shortening the name of this macro to just `swagger_schema`.
In the spec, the `definitions` object (http://swagger.io/specification/#definitionsObject) is the container for multiple schema objects (http://swagger.io/specification/#schemaObject).